### PR TITLE
feat(elevation_map_loader): reduce memory usage of elevation_map_loader

### PR DESCRIPTION
**Pull request by Shin-kyoto**
## Description

- When you use large pointcloud map, elevation_map_loader use so much amount of memory and keep it after making elevation_map. So in this PR, remove unused object from member of `ElevationMapLoaderNode`.
- Before PR, 10GB memory is allocated after generating elevation_map. (Elevation_map is generated in the memory peak with 13+αGB.)
before_PR_

- After PR, 1+αGB memory is allocated after generating elevation_map. (Elevation_map is generated in the memory peak with 12GB.)
afterPR

## Related links
JIRA link: [TIERIV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-24434)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/elevation_map_loader/include/elevation_map_loader/elevation_map_loader_node.hpp [hunk 1]
```diff
@@ -73,7 +73,8 @@ class ElevationMapLoaderNode : public rclcpp::Node
   void publish();
   void createElevationMap();
   void setVerbosityLevelToDebugIfFlagSet();
-  void createElevationMapFromPointcloud();
+  void createElevationMapFromPointcloud(
+    const pcl::shared_ptr<grid_map::GridMapPclLoader> & grid_map_pcl_loader_);
```
Comment (kosuke55) [inline perception/elevation_map_loader/include/elevation_map_loader/elevation_map_loader_node.hpp:77]:
```suggestion
  void createElevationMapFromPointcloud(
    const pcl::shared_ptr<grid_map::GridMapPclLoader> & grid_map_pcl_loader);
```

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 2]
```diff
@@ -177,6 +174,11 @@ void ElevationMapLoaderNode::onVectorMap(
 
 void ElevationMapLoaderNode::createElevationMap()
 {
+  auto grid_map_logger = rclcpp::get_logger("grid_map_logger");
+  grid_map_logger.set_level(rclcpp::Logger::Level::Error);
+  pcl::shared_ptr<grid_map::GridMapPclLoader> grid_map_pcl_loader_ =
+    pcl::make_shared<grid_map::GridMapPclLoader>(grid_map_logger);
+  grid_map_pcl_loader_->loadParameters(this->param_file_path);
```
Comment (kosuke55) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:181]:
```suggestion
  pcl::shared_ptr<grid_map::GridMapPclLoader> grid_map_pcl_loader =
    pcl::make_shared<grid_map::GridMapPclLoader>(grid_map_logger);
  grid_map_pcl_loader->loadParameters(param_file_path_);
```

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 3]
```diff
@@ -187,15 +189,16 @@ void ElevationMapLoaderNode::createElevationMap()
   } else {
     grid_map_pcl_loader_->setInputCloud(data_manager_.map_pcl_ptr_);
   }
-  createElevationMapFromPointcloud();
+  createElevationMapFromPointcloud(grid_map_pcl_loader_);
```
Comment (kosuke55) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:192]:
```suggestion
  createElevationMapFromPointcloud(grid_map_pcl_loader);
```

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 4]
```diff
@@ -187,15 +189,16 @@ void ElevationMapLoaderNode::createElevationMap()
...
   elevation_map_ = grid_map_pcl_loader_->getGridMap();
```
Comment (kosuke55) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:193]:
```suggestion
  elevation_map_ = grid_map_pcl_loader->getGridMap();
```

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 5]
```diff
@@ -187,15 +189,16 @@ void ElevationMapLoaderNode::createElevationMap()
...
   if (use_inpaint_) {
     inpaintElevationMap(inpaint_radius_);
   }
   saveElevationMap();
 }
 
-void ElevationMapLoaderNode::createElevationMapFromPointcloud()
+void ElevationMapLoaderNode::createElevationMapFromPointcloud(
+  const pcl::shared_ptr<grid_map::GridMapPclLoader> & grid_map_pcl_loader_)
```
Comment (kosuke55) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:201]:
```suggestion
void ElevationMapLoaderNode::createElevationMapFromPointcloud(
  const pcl::shared_ptr<grid_map::GridMapPclLoader> & grid_map_pcl_loader)
```

File: perception/elevation_map_loader/include/elevation_map_loader/elevation_map_loader_node.hpp [hunk 6]
```diff
@@ -97,7 +98,7 @@ class ElevationMapLoaderNode : public rclcpp::Node
   bool use_inpaint_;
   float inpaint_radius_;
   bool use_elevation_map_cloud_publisher_;
-  pcl::shared_ptr<grid_map::GridMapPclLoader> grid_map_pcl_loader_;
+  std::string param_file_path;
```
Comment (kosuke55) [inline perception/elevation_map_loader/include/elevation_map_loader/elevation_map_loader_node.hpp:101]:
```suggestion
  std::string param_file_path_;
```

Comment (kosuke55) [reply perception/elevation_map_loader/include/elevation_map_loader/elevation_map_loader_node.hpp:101]: or not making a member, using  `this->declare_parameter("param_file_path", "path_default");` in `createElevationMap()` may be good

File: perception/elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 7]
```diff
@@ -50,7 +50,7 @@ ElevationMapLoaderNode::ElevationMapLoaderNode(const rclcpp::NodeOptions & optio
 : Node("elevation_map_loader", options)
 {
   layer_name_ = this->declare_parameter("map_layer_name", std::string("elevation"));
-  std::string param_file_path = this->declare_parameter("param_file_path", "path_default");
+  this->param_file_path = this->declare_parameter("param_file_path", "path_default");
```
Comment (kosuke55) [inline perception/elevation_map_loader/src/elevation_map_loader_node.cpp:53]:
```suggestion
  param_file_path_ = this->declare_parameter("param_file_path", "path_default");
```

Comment (kosuke55) [reply perception/elevation_map_loader/src/elevation_map_loader_node.cpp:53]: it is better to align with other variables.

**Comment by Shin-kyoto** [comment]
@kosuke55 
Thank you for your review! I fixed variables' name.

**Review by kosuke55 — APPROVED** [review]
thanks very much, I did not test but the code looks good.

**Comment by Shin-kyoto** [comment]
> thanks very much, I did not test but the code looks good.
- I checked code by generating elevation map with [sample_map](https://drive.google.com/file/d/1A-8BvYRX3DhSzkAnOcGWFw5T30xTlwZI/view).
- Checked by zooming elevation_map and found no broken points.
image
- No large holes in elevation_map from top view.
image
- The elevation_map does not deviate significantly from the point cloud.
image

- And this PR does not modify main algorithm, so it is enough to check for major breakdowns. I visualized elevation_map and checked it, and there was no problem. So I merge this PR.

**Pull request merged**